src/main/java/uk/gov/companieshouse/api/accounts/validation/CreditorsAfterOneYearValidator.java

```diff
@@ -0,0 +1,340 @@
+
+        if (!isPreviousPeriodBalanceSheetCreditorsAfterOneYearNull(previousPeriodBalanceSheet) &&
+            (!isCreditorsAfterOneYearNotePreviousTotalNull(creditorsAfterOneYear)) &&
+            (!creditorsAfterOneYear.getPreviousPeriod().getTotal().equals(
+                previousPeriodBalanceSheet.getOtherLiabilitiesOrAssets().getCreditorsAfterOneYear()))) {
+
+            addError(errors, currentBalanceSheetNotEqual, CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void checkIsPreviousBalanceNullAndNoteNot(Errors errors, CreditorsAfterOneYear creditorsAfterOneYear,
+                                                      BalanceSheet previousPeriodBalanceSheet) {
+
+        if (isPreviousPeriodBalanceSheetCreditorsAfterOneYearNull(previousPeriodBalanceSheet) &&
+            (!isCreditorsAfterOneYearNotePreviousTotalNull(creditorsAfterOneYear))) {
+
+            addError(errors, previousBalanceSheetNotEqual, CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void checkIfPreviousNoteIsNullAndBalanceNot(Errors errors, CreditorsAfterOneYear creditorsAfterOneYear,
+                                                        BalanceSheet previousPeriodBalanceSheet) {
+
+        if (isCreditorsAfterOneYearNotePreviousTotalNull(creditorsAfterOneYear) &&
+            (!isPreviousPeriodBalanceSheetCreditorsAfterOneYearNull(previousPeriodBalanceSheet))) {
+
+            addError(errors, previousBalanceSheetNotEqual, CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private boolean isCreditorsAfterOneYearNotePreviousTotalNull(CreditorsAfterOneYear creditorsAfterOneYear) {
+        return creditorsAfterOneYear.getPreviousPeriod() == null ||
+            creditorsAfterOneYear.getPreviousPeriod().getTotal() == null;
+    }
+
+    private boolean isPreviousPeriodBalanceSheetCreditorsAfterOneYearNull(BalanceSheet previousPeriodBalanceSheet) {
+        return previousPeriodBalanceSheet == null ||
+            previousPeriodBalanceSheet.getOtherLiabilitiesOrAssets() == null ||
+            previousPeriodBalanceSheet.getOtherLiabilitiesOrAssets().getCreditorsAfterOneYear() == null;
+    }
+
+    private BalanceSheet getCurrentPeriodBalanceSheet(HttpServletRequest request,
+                                                      String companyAccountsId) throws DataException {
+
+        String currentPeriodId = currentPeriodService.generateID(companyAccountsId);
+
+        ResponseObject<uk.gov.companieshouse.api.accounts.model.rest.CurrentPeriod> currentPeriodResponseObject;
+
+        try {
+            currentPeriodResponseObject = currentPeriodService.findById(currentPeriodId, request);
+        } catch (MongoException me) {
+            throw new DataException(me.getMessage(), me);
+        }
+
+        if (currentPeriodResponseObject != null && currentPeriodResponseObject.getData() != null) {
```

Comment:
Rather than making this check and having 2 return statements, you just make use of `Optional`'s.
Something like:
```
return Optional.ofNullable(currentPeriodResponseObject)
                          .map(ResponseObject::getData)
                          .map(CurrentPeriod::getBalanceSheet)
                          .orElse(null);
```

Reply:
refactored this one and previousPeriodResponseObject